File: src/mcp/clear_collection.ts
import { QdrantClient } from "@qdrant/js-client-rest";
import dotenv from "dotenv";

dotenv.config();

const client = new QdrantClient({
  url: process.env.QDRANT_URL!,
  apiKey: process.env.QDRANT_KEY!,
});

export async function clearCollection() {
  const collectionName = "document-qna";

  try {
    const response = await client.delete(collectionName, {
      filter: {
        must: [], // Match all documents
      },
    });

    return { success: true, message: "Collection cleared", response };
  } catch (error: any) {
    console.error("❌ Error while clearing collection:", error);
    return {
      success: false,
      message: "Failed to clear collection",
      error: error?.message || error,
    };
  }
}

File: src/mcp/clear_collection.py
import os
from qdrant_client import QdrantClient
from qdrant_client.models import Filter


def clear_collection() -> dict:
    """Clear all documents from the Qdrant collection."""
    qdrant_url = os.getenv("QDRANT_URL")
    qdrant_key = os.getenv("QDRANT_KEY")

    qdrant = QdrantClient(url=qdrant_url, api_key=qdrant_key)
    collection_name = "document-qna"

    try:
        response = qdrant.delete(
            collection_name=collection_name,
            points_selector=Filter(must=[])  # Match all documents
        )

        return {"success": True, "message": "Collection cleared", "response": str(response)}
    except Exception as error:
        print(f"❌ Error while clearing collection: {error}")
        return {
            "success": False,
            "message": "Failed to clear collection",
            "error": str(error)
        }
